### PR TITLE
Rename Container to Step

### LINKED ISSUE #1736: [Feature Request]: Terminology for Pipeline Logs
### Feature description

Follow up on https://github.com/opendatahub-io/odh-dashboard/issues/1247

`Containers` is our UI focus right now. 
`Steps` is Tekton's run state.

Should we look to talk more about steps and less about pure log containers. We may want to look at improving log-based naming? We talk about pods in errors because K8s talks about it. We could look to try and not show errors flat footed -- but it may be difficult.

### Describe alternatives you've considered

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by manaswinidas**
Fixes #1736 

## Description

Renamed "Container" to "Steps".

cc @yih-wang 

Screenshot 2023-10-25 at 4 59 12 PM

## How Has This Been Tested?

1. Go to Global Pipelines Run page and select any Triggered run.
2. Click on any node -> A side drawer will appear from the right containing the "Logs" tab.
3. You can see "Steps" instead of "Container" beside the steps-dropdown.

## Request review criteria:

Self checklist (all need to be checked):
- [X] The developer has manually tested the changes and verified that the changes work
- [X] Commits have been squashed into descriptive, self-contained units of work (e.g. 'WIP' and 'Implements feedback' style messages have been removed)
- [X] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has added tests or explained why testing cannot be added (unit tests & storybook for related changes)

If you have UI changes: 

- [X] Included any necessary screenshots or gifs if it was a UI change.
- [X] Included tags to the UX team if it was a UI/UX change (find relevant UX in the [SMEs](https://github.com/opendatahub-io/odh-dashboard/tree/main/docs/smes.md) section).

After the PR is posted & before it merges:
- [ ] The developer has tested their solution on a cluster by using the image produced by the PR to `main`

### CODE REVIEW

File: frontend/src/concepts/pipelines/content/pipelinesDetails/pipelineRun/runLogs/LogsTab.tsx [hunk 1]
```diff
@@ -103,7 +103,9 @@ const LogsTabForPodName: React.FC<{ podName: string }> = ({ podName }) => {
               <SplitItem>
                 <Bullseye>
                   <TextContent>
-                    <TextListItem component={TextListItemVariants.dt}>Container</TextListItem>
+                    <TextListItem component={TextListItemVariants.dt}>
+                      {podContainers.length === 0 ? 'Step' : 'Steps'}
```
Comment (manaswinidas) [inline frontend/src/concepts/pipelines/content/pipelinesDetails/pipelineRun/runLogs/LogsTab.tsx:107]: @yih-wang I saw in the [mocks](https://www.figma.com/proto/rJjOtl3nxpW6BAcl0MZr4y/Pipeline-log-viewer?page-id=62%3A1895&type=design&node-id=260-2386&viewport=664%2C-1095%2C0.23&t=80ExnogduAcwD3kx-1&scaling=min-zoom&starting-point-node-id=260%3A2386) that we see "Steps" when there are multiple steps and "Step" when there is one. Is this intentional? Just confirming coz I didn't see any comments regarding this in the mocks.

Comment (andrewballantyne) [reply frontend/src/concepts/pipelines/content/pipelinesDetails/pipelineRun/runLogs/LogsTab.tsx:107]: Interesting... I always assumed this was "Step <blank>" "Container <blank>"... is this intended as a label for the dropdown? "Steps" "Containers" as in "there are multiple involve here"?

Comment (yih-wang) [reply frontend/src/concepts/pipelines/content/pipelinesDetails/pipelineRun/runLogs/LogsTab.tsx:107]: @andrewballantyne No, it's not intentional. The only reason for the difference is my bad English... I believe the label should stay consistent, whether there's one or more options in the dropdown. I have no idea about whether to use the singular or plural form in this context. @kaedward, could you help on this matter?

Comment (yih-wang) [reply frontend/src/concepts/pipelines/content/pipelinesDetails/pipelineRun/runLogs/LogsTab.tsx:107]: But I guess the singular form (Step) is the answer according to what we have for the other dropdown labels around the console.

Comment (andrewballantyne) [reply frontend/src/concepts/pipelines/content/pipelinesDetails/pipelineRun/runLogs/LogsTab.tsx:107]: Ah gotcha... thanks for the clarification. @manaswinidas implement just singular.

**Pull request merged**